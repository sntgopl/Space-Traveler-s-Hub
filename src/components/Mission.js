import PropTypes from 'prop-types';
import { Button, Badge } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../redux/missions/missions';

const Mission = ({
  name, description, id, reserved,
}) => {
  const dispatch = useDispatch();
  return (
    <tbody>
      <tr>
        <td>
          {name}
        </td>
        <td>{description}</td>
        <td>
          {!reserved && (
            <Badge bg="secondary">NOT A MEMBER</Badge>
          )}
          {reserved && (
            <Badge bg="primary">Active Member</Badge>
          )}
        </td>
        <td>
          {!reserved && (
            <Button
              onClick={() => dispatch(joinMission(id))}
              variant="outline-secondary"
            >
              Join Mission
            </Button>
          )}
          {reserved && (
            <Button
              onClick={() => dispatch(leaveMission(id))}
              variant="outline-danger"
            >
              Leave Mission
            </Button>
          )}

        </td>
      </tr>
    </tbody>

  );
};

Mission.propTypes = {
  name: PropTypes.node.isRequired,
  description: PropTypes.node.isRequired,
  id: PropTypes.node.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Mission;
