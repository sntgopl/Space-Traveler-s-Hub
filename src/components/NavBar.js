import { NavLink } from 'react-router-dom';
import image from '../images/planet.png';
import '../stylesheets/NavBar.css';

const NavBar = () => (
  <nav>
    <div className="navTitle">
      <img src={image} alt="planet" />
      <h1>Space Traveler`s Hub</h1>
    </div>
    <div className="navList">
      <ul>
        <NavLink className={({ isActive }) => (isActive ? 'activeLink' : '')} to="/">Rockets</NavLink>
        <NavLink className={({ isActive }) => (isActive ? 'activeLink' : '')} to="/missions">Missions</NavLink>
        <NavLink className={({ isActive }) => (isActive ? 'activeLink' : '')} to="/profile">My Profile</NavLink>
      </ul>
    </div>
  </nav>
);

export default NavBar;
