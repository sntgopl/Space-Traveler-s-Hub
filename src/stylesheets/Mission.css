table {
  margin-left: 5vw;
  max-width: 90vw;
}

td {
  min-width: 10vw;
}

th {
  min-width: 10vw;
}
