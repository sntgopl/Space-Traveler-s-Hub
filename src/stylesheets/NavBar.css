nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 90%;
  margin: auto;
  border-bottom: 1px solid #ccc;
}

.navTitle {
  display: flex;
  align-items: center;
  gap: 10px;
  margin: 10px 0;
}

.navTitle img {
  width: 70px;
}

.navTitle h1 {
  margin: 0;
}

.navList ul {
  display: flex;
  margin: 0;
}

.navList ul :nth-child(even) {
  border-right: 1px solid #000;
}

.navList ul > * {
  text-decoration: none;
  padding: 0 10px;
}

.activeLink {
  opacity: 0.5;
  border-bottom: 1px solid;
}
