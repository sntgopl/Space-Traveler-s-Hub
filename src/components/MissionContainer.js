import Table from 'react-bootstrap/Table';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getMission } from '../redux/missions/missions';
import Mission from './Mission';

const MissionContainer = () => {
  const missions = useSelector((state) => state.mission);
  const dispatch = useDispatch();
  useEffect(() => {
    if (!missions.length) {
      dispatch(getMission());
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <Table striped bordered>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th>Function</th>
        </tr>
      </thead>
      {missions.map((mission) => (
        <Mission
          key={mission.id}
          id={mission.id}
          name={mission.name}
          description={mission.description}
          reserved={mission.reserved}
        />
      ))}
    </Table>
  );
};

export default MissionContainer;
