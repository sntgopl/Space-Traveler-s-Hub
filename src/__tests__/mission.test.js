import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Table from 'react-bootstrap/Table';
import Mission from '../components/Mission';
import MissionContainer from '../components/MissionContainer';
import NavBar from '../components/NavBar';
import store from '../redux/configureStore';

describe('render Mission', () => {
  it('should render', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Table>
            <Mission
              key={1}
              id={1}
              name="Mission test"
              description="first mission"
              reserved={false}
            />
          </Table>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('render Mission', () => {
  it('should render a mission', () => {
    render(
      <Provider store={store}>
        <Table>
          <Mission
            key={1}
            id={1}
            name="Mission test"
            description="first mission"
            reserved={false}
          />
        </Table>
      </Provider>,
    );
    expect(screen.getByText('Join Mission')).toBeInTheDocument();
  });

  it('should render a mission', () => {
    render(
      <Provider store={store}>
        <Table>
          <Mission
            key={1}
            id={1}
            name="Mission test"
            description="first mission"
            reserved
          />
        </Table>
      </Provider>,
    );
    expect(screen.getByText('Leave Mission')).toBeInTheDocument();
  });
});

describe('render Mission Container', () => {
  it('should render', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MissionContainer />
        </Provider>,
      )
      .toJSON();
    expect(tree).toBeDefined();
  });
});

describe('render Nav Bar', () => {
  it('should render Nav Bar', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
