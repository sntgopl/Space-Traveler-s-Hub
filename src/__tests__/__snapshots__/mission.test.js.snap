// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Mission should render 1`] = `
<table
  className="table"
>
  <tbody>
    <tr>
      <td>
        Mission test
      </td>
      <td>
        first mission
      </td>
      <td>
        <span
          className="badge bg-secondary"
        >
          Non-Active
        </span>
      </td>
      <td>
        <button
          className="btn btn-outline-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`render Nav Bar should render Nav Bar 1`] = `
<nav>
  <div
    className="navTitle"
  >
    <img
      alt="planet"
      src="planet.png"
    />
    <h1>
      Space Traveler\`s Hub
    </h1>
  </div>
  <div
    className="navList"
  >
    <ul>
      <a
        aria-current="page"
        className="activeLink"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <a
        className=""
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </ul>
  </div>
</nav>
`;
